announce.go

```diff
@@ -269,13 +307,53 @@ func (a *Announce) nodeContactor() {
 			break
 		}
 		go txRes.run()
+		time.Sleep(time.Millisecond * 100)
+	}
+}
+
+func (a *Announce) logic(tx *stm.Tx) interface{} {
+	inFlightGetPeers := atomic.LoadInt64(&a.inFlightGetPeers)
+	var mode dhtMode
+	nowMode := tx.Get(a.mode).(dhtMode)
+	mode = nowMode
+	if nowMode == get_peers && inFlightGetPeers >= 3 {
+		a.server.logger().Printf("dht Changed mode to: get_peers_stall")
+		mode = get_peers_stall
+	}
+	if nowMode == get_peers_stall && inFlightGetPeers < 1 {
+		mode = get_peers
+		a.server.logger().Printf("dht Changed mode to: from get_peers_stall to get_peers")
+	}
+
+	l := tx.Get(a.pendingAnnouncePeers).(stmutil.List)
+	if l.Len() > 8 {
+		go func() {
+			time.Sleep(time.Minute * 15)
+			stm.AtomicSet(a.mode, get_peers)
+			a.server.logger().Printf("dht Changed mode to: get_peers")
+		}()
+		mode = announce_peer
+		a.server.logger().Printf("dht Changed mode to: announce_peer")
+		// Is it the best way to get a sorted list of good node?
+		a.traversal = newTraversal(a.infoHash)
+		startAddrs, _ := a.server.traversalStartingNodes()
+		for _, n := range startAddrs {
+			stm.Atomically(a.pendContact(n))
+		}
+	}
```

Comment:
Should I reload the whole node table or is it better to keep one linked to the whole announce live?

Reply:
I'm not sure why the starting nodes are reloaded here.